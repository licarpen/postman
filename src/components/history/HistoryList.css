.HistoryList{
  grid-area: historyList;
  font-family: Arial, Helvetica, sans-serif;
  
  ul {
    list-style: none;
  }

  li {
    border: 1px black solid;
    padding: 10px;
    margin: 5px;
  }

}