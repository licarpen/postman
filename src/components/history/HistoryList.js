import React from 'react';
import PropTypes from 'prop-types';
import HistoryItem from './HistoryItem';
import styles from './HistoryList.css';

const HistoryList = ({ history, onClick }) => {
  const historyElements = history.map(historyItem => (
    <li key={historyItem.method - historyItem.url}>
      <HistoryItem {...historyItem} onClick={onClick}/>
    </li>
  ));

  return (
    <div className={styles.HistoryList}>
      <h2>History</h2>
      <ul >
        {historyElements}
      </ul>
    </div>
  );
};

HistoryList.propTypes = {
  onClick: PropTypes.func.isRequired,
  history: PropTypes.arrayOf(PropTypes.shape({
    url: PropTypes.string.isRequired,
    method: PropTypes.string.isRequired,
    body: PropTypes.string.isRequired
  }))
};

export default HistoryList;
