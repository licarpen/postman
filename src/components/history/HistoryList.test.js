import React from 'react';
import { shallow } from 'enzyme';
import HistoryList from './HistoryList';

describe('HistoryList component', () => {
  it('matches a snapshot', () => {
    const historyArray = [{ url: 'sampleurl.com', method: 'POST', body: 'sample body' }];
    const wrapper = shallow(<HistoryList onClick={() => {}} history={historyArray} />);
    expect(wrapper).toMatchSnapshot();
  });
});
