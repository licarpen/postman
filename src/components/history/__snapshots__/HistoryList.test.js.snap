// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoryList component matches a snapshot 1`] = `
<div
  className="HistoryList"
>
  <h2>
    History
  </h2>
  <ul>
    <li
      key="NaN"
    >
      <HistoryItem
        body="sample body"
        method="POST"
        onClick={[Function]}
        url="sampleurl.com"
      />
    </li>
  </ul>
</div>
`;
